apps/captain_fact/lib/captain_fact/videos/captions_fetcher.ex

```diff
@@ -0,0 +1,40 @@
+defmodule CaptainFact.Videos.CaptionsFetcher do
+  @moduledoc """
+  Fetch captions for videos.
+  """
+
+  if Application.get_env(:db, :env) == :test do
+    def fetch_video_captions({"__TEST__", _youtube_id}, locale) do
+      {:ok,
+       %{content: "__TEST-CONTENT__", format: "xml", locale: locale, is_auto_generated: true}}
+    end
+  end
+
+  @doc """
+  Fetches video captions from YouTube.
+  """
+  def fetch_video_captions({"youtube", youtube_id}, locale) do
+    with {:ok, content} <- fetch_captions_content(youtube_id, locale) do
+      {:ok, %{content: content, format: "xml", locale: locale, is_auto_generated: true}}
```

Comment:
Not sure about `is_auto_generated`, if there is no use-case it could be removed?

Reply:
Indeed. The idea was to be able to update subtitles if users-generated one were provided by YouTube after we already fetched automatic transcript, but if we can't get that information from YouTube API then we should discard it.